src/data/lazy_list2.lean

```diff
@@ -87,6 +87,12 @@ def init {α} : lazy_list α → lazy_list α
   | (lazy_list.cons _ _) := lazy_list.cons x (init xs')
   end
 
```

Comment:
Why is this file called `lazy_list2.lean` in the first place?

Could you add a module doc while we're at it?

Reply:
`data.lazy_list` is one of the few files in core that is not in `init`. If the file was named `lazy_list.lean`, the name `data.lazy_list` would resolve to both files, and Lean behaves weirdly when that happens. `data.vector2` is the same way. (That said, there are other ways to resolve the situation, this is not an endorsement.)

Reply:
@digama0 I thought you were the one who advocated for that naming scheme. I would favor `data.lazy_list.basic` and `data.vector.basic`

Reply:
I'm the one who originally implemented it, but that doesn't mean I find it a great solution. Times have changed since `data.vector2` wad first written; at this point I think the better solution is just to remove `vector` and `lazy_list` from core and merge the files.

Reply:
We may have the luxury to consolidate that kind of module because we maintain both mathlib and Lean 3.5c but I feel like this answers avoids an important part of the question. If you write a library where you define `foo_bar` and I import that library and decide to prove things about `foo_bar` or add new features, I don't find that that having `data.foo_bar` and `data.foo_bar2` is a nice solution. And my constructions might not be interesting for you to have in your library. I think a convention like `data.foo_bar.simon's_library` or like `simon's_library.data.foo_bar` would serve us better if people are to publish Lean code outside of mathlib.

Reply:
How about here we just rename to `data.lazy_list.basic`, and add a TODO in the module doc that Simon is going to write that we should remove `data.lazy_list` from core at some point.

Reply:
The problem here is that both mathlib and core are purporting to be "the" standard library and so feel no need to further scope their import paths. For a third party library you would probably put everything under an additional import level (like `my_library.data.lazy_list`) and then there is no problem. This is already the case for the most part between core and mathlib because almost everything in core is in the `init` folder, while `mathlib` has no corresponding `init` folder. It is only these non-init files that cause a problem.